artemis-server/src/main/java/org/apache/activemq/artemis/core/settings/impl/HierarchicalObjectRepository.java

```diff
@@ -29,13 +29,13 @@
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
-import java.util.regex.Pattern;
 
 import org.apache.activemq.artemis.core.config.WildcardConfiguration;
 import org.apache.activemq.artemis.core.server.ActiveMQServerLogger;
 import org.apache.activemq.artemis.core.settings.HierarchicalRepository;
 import org.apache.activemq.artemis.core.settings.HierarchicalRepositoryChangeListener;
 import org.apache.activemq.artemis.core.settings.Mergeable;
+import org.apache.commons.lang3.StringUtils;
```

Comment:
My only concern here is that commons.lang is an indirect dependency. if we use this we need to make a direct declaration.

mvn dependency:tree would give us this:

```
[INFO] --- maven-dependency-plugin:3.1.1:tree (default-cli) @ artemis-server ---
[INFO] org.apache.activemq:artemis-server:jar:2.17.0-SNAPSHOT
[INFO] +- org.jboss.logging:jboss-logging-processor:jar:2.2.0.Final:provided (optional)
[INFO] |  \- org.jboss.jdeparser:jdeparser:jar:2.0.3.Final:provided (optional)
[INFO] +- org.jboss.logging:jboss-logging-annotations:jar:2.2.0.Final:provided (optional)
[INFO] +- com.google.errorprone:error_prone_core:jar:2.4.0:provided
[INFO] |  +- com.google.errorprone:error_prone_annotation:jar:2.4.0:provided
[INFO] |  +- com.google.errorprone:error_prone_type_annotations:jar:2.4.0:provided
[INFO] |  +- com.google.errorprone:error_prone_check_api:jar:2.4.0:provided
[INFO] |  |  +- com.googlecode.java-diff-utils:diffutils:jar:1.3.0:provided
[INFO] |  |  +- com.github.kevinstern:software-and-algorithms:jar:1.0:provided
[INFO] |  |  \- com.github.ben-manes.caffeine:caffeine:jar:2.8.0:provided
[INFO] |  +- org.pcollections:pcollections:jar:2.1.2:provided
[INFO] |  +- com.google.auto:auto-common:jar:0.10:provided
[INFO] |  +- com.google.code.findbugs:jFormatString:jar:3.0.0:provided
[INFO] |  +- com.google.code.findbugs:jsr305:jar:3.0.0:compile
[INFO] |  +- org.checkerframework:dataflow-shaded:jar:3.1.2:provided
[INFO] |  +- com.google.errorprone:javac:jar:9+181-r4173-1:provided
[INFO] |  +- com.google.auto.value:auto-value-annotations:jar:1.7:provided
[INFO] |  +- com.google.errorprone:error_prone_annotations:jar:2.4.0:compile
[INFO] |  +- com.google.protobuf:protobuf-java:jar:3.4.0:provided
[INFO] |  +- com.google.auto.service:auto-service-annotations:jar:1.0-rc6:provided
[INFO] |  \- org.threeten:threeten-extra:jar:1.5.0:provided
[INFO] +- com.google.guava:guava:jar:30.1-jre:compile
[INFO] |  +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO] |  +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[INFO] |  +- org.checkerframework:checker-qual:jar:3.5.0:compile
[INFO] |  \- com.google.j2objc:j2objc-annotations:jar:1.3:compile
[INFO] +- org.jboss.logging:jboss-logging:jar:3.4.0.Final:compile
[INFO] +- org.jboss.logmanager:jboss-logmanager:jar:2.1.10.Final:compile
[INFO] |  \- org.wildfly.common:wildfly-common:jar:1.5.2.Final:compile
[INFO] +- org.apache.activemq:artemis-commons:jar:2.17.0-SNAPSHOT:compile
[INFO] +- org.apache.activemq:artemis-selector:jar:2.17.0-SNAPSHOT:compile
[INFO] +- org.apache.activemq:artemis-journal:jar:2.17.0-SNAPSHOT:compile
[INFO] +- org.apache.activemq:artemis-jdbc-store:jar:2.17.0-SNAPSHOT:compile
[INFO] |  \- org.apache.commons:commons-dbcp2:jar:2.7.0:compile
[INFO] |     \- org.apache.commons:commons-pool2:jar:2.7.0:runtime
[INFO] +- org.apache.activemq:artemis-core-client:jar:2.17.0-SNAPSHOT:compile
[INFO] |  +- org.jgroups:jgroups:jar:3.6.13.Final:compile
[INFO] |  +- org.apache.johnzon:johnzon-core:jar:0.9.5:compile
[INFO] |  +- io.netty:netty-handler-proxy:jar:4.1.51.Final:compile
[INFO] |  \- io.netty:netty-codec-socks:jar:4.1.51.Final:compile
[INFO] +- org.apache.activemq:activemq-artemis-native:jar:1.0.2:compile
[INFO] +- org.apache.activemq:artemis-core-client:test-jar:tests:2.17.0-SNAPSHOT:test
[INFO] +- org.jctools:jctools-core:jar:2.1.2:compile
[INFO] +- io.netty:netty-buffer:jar:4.1.51.Final:compile
[INFO] +- io.netty:netty-transport:jar:4.1.51.Final:compile
[INFO] |  \- io.netty:netty-resolver:jar:4.1.51.Final:compile
[INFO] +- io.netty:netty-codec-http:jar:4.1.51.Final:compile
[INFO] +- io.netty:netty-common:jar:4.1.51.Final:compile
[INFO] +- io.netty:netty-handler:jar:4.1.51.Final:compile
[INFO] +- io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.51.Final:compile
[INFO] |  \- io.netty:netty-transport-native-unix-common:jar:4.1.51.Final:compile
[INFO] +- io.netty:netty-transport-native-kqueue:jar:osx-x86_64:4.1.51.Final:compile
[INFO] +- io.netty:netty-codec:jar:4.1.51.Final:compile
[INFO] +- commons-beanutils:commons-beanutils:jar:1.9.4:compile
[INFO] |  +- commons-logging:commons-logging:jar:1.2:compile
[INFO] |  \- commons-collections:commons-collections:jar:3.2.2:compile
[INFO] +- org.apache.commons:commons-configuration2:jar:2.7:compile
[INFO] |  +- org.apache.commons:commons-lang3:jar:3.0:compile
[INFO] |  \- org.apache.commons:commons-text:jar:1.8:compile
[INFO] +- io.micrometer:micrometer-core:jar:1.6.1:compile
[INFO] |  +- org.hdrhistogram:HdrHistogram:jar:2.1.12:compile
[INFO] |  \- org.latencyutils:LatencyUtils:jar:2.0.3:runtime
[INFO] +- commons-io:commons-io:jar:2.6:compile
[INFO] +- org.apache.geronimo.specs:geronimo-json_1.0_spec:jar:1.0-alpha-1:compile
[INFO] +- junit:junit:jar:4.11:test
[INFO] |  \- org.hamcrest:hamcrest-core:jar:1.3:test
[INFO] +- org.apache.directory.server:apacheds-test-framework:jar:2.0.0-M15:test
[INFO] |  +- org.apache.directory.server:apacheds-core-api:jar:2.0.0-M15:test
[INFO] |  |  +- commons-lang:commons-lang:jar:2.6:test
[INFO] |  |  +- org.apache.directory.server:apacheds-core-constants:jar:2.0.0-M15:test
[INFO] |  |  +- org.apache.directory.api:api-i18n:jar:1.0.0-M20:test
[INFO] |  |  +- org.apache.directory.api:api-asn1-api:jar:1.0.0-M20:test
[INFO] |  |  +- org.apache.directory.api:api-ldap-client-api:jar:1.0.0-M20:test
[INFO] |  |  |  \- commons-pool:commons-pool:jar:1.6:test
[INFO] |  |  +- org.apache.directory.api:api-ldap-extras-aci:jar:1.0.0-M20:test
[INFO] |  |  +- org.apache.directory.api:api-ldap-extras-util:jar:1.0.0-M20:test
[INFO] |  |  \- net.sf.ehcache:ehcache-core:jar:2.4.4:test
[INFO] |  +- org.apache.directory.server:apacheds-i18n:jar:2.0.0-M15:test
[INFO] |  +- org.apache.directory.server:apacheds-protocol-kerberos:jar:2.0.0-M15:test
[INFO] |  |  +- org.apache.directory.server:apacheds-kerberos-codec:jar:2.0.0-M15:test
[INFO] |  |  \- org.apache.directory.server:apacheds-core-shared:jar:2.0.0-M15:test
[INFO] |  +- org.apache.directory.server:apacheds-protocol-ldap:jar:2.0.0-M15:test
[INFO] |  |  +- org.apache.directory.jdbm:apacheds-jdbm1:bundle:2.0.0-M2:test
[INFO] |  |  +- org.apache.directory.api:api-asn1-ber:jar:1.0.0-M20:test
[INFO] |  |  +- org.apache.directory.api:api-ldap-extras-codec-api:jar:1.0.0-M20:test
[INFO] |  |  +- org.apache.directory.api:api-ldap-extras-codec:jar:1.0.0-M20:test
[INFO] |  |  +- org.apache.directory.api:api-ldap-extras-sp:jar:1.0.0-M20:test
[INFO] |  |  \- bouncycastle:bcprov-jdk15:jar:140:test
[INFO] |  +- org.apache.directory.server:apacheds-protocol-shared:jar:2.0.0-M15:test
[INFO] |  +- org.apache.directory.api:api-ldap-codec-standalone:jar:1.0.0-M20:test
```

Reply:
Ok just changed the code to a similar way it was before but fixed the for to reflect the right order.
So the import is no longer needed